File: CocktailsProject/app/src/main/java/com/example/cocktailsproject/MainActivity.java
package com.example.cocktailsproject;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Observer;

import android.os.Bundle;

import com.example.cocktailsproject.datasource.APIManager;
import com.example.cocktailsproject.models.Cocktail;
import com.example.cocktailsproject.ui.main.MainFragment;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);
    }
}

File: Lecture Files/Android/Files/MainActivity.kt
package com.example.intents

import android.content.Intent
import android.content.IntentFilter
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }

    fun moveToNextScreen() {
        // Intent with a result expected
        val intent = Intent(this, SecondActivity::class.java)
        startActivity(intent)
    }

    fun moveToNextScreenWithResult() {
        // Intent with a result expected
        val intent = Intent(this, SecondActivity::class.java)
        val intentLauncher = registerForActivityResult(
            ActivityResultContracts.StartActivityForResult()
        ) {
            // here we receive result from the intent
        }
        // launch the intent with the result launcher
        intentLauncher.launch(intent)
    }

    fun implicitIntents() {
        // implicit intent to send data to other app
        val sendIntent: Intent = Intent().apply {
            action = Intent.ACTION_SEND
            putExtra(Intent.EXTRA_TEXT, "This is my text to send.")
            type = "text/plain"
        }
        // lets the user choose an app to send the data to (ex: send SMS)
        val shareIntent = Intent.createChooser(sendIntent, null)
        startActivity(shareIntent)
    }
}